hack/cmd_util.sh

```diff
@@ -156,9 +169,10 @@ function os::cmd::internal::expect_exit_code_run_grep() {
 	local test_eval_func=${4:-os::cmd::internal::success_func}
 
 	os::cmd::internal::init_tempdir
+	os::test::junit::declare_test_start
```

Comment:
If I nest `expect_success` under a `try_until_success`, will you correctly collapse the repeated test in the junit output and report success for it?